client-config/src/main/java/com/palantir/conjure/java/client/config/HttpsProxy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.client.config;
+
+import java.net.Proxy;
+import java.net.SocketAddress;
+
+public class HttpsProxy extends Proxy {
```

Comment:
I might suggest a slight change in the public API portion of this change such that we expose a bit less to consumers. Instead of exposing a public implementation of `Proxy`, what if we added a public utility class along these lines?

```java
public final class HttpsProxies {

    public static Proxy create(InetSocketAddress address) {
        // TODO
    }

    public static boolean isHttps(Proxy proxy) {
        // TODO
    }

    private HttpsProxies() {}
    
    private static final class HttpsProxy extends Proxy {
        // TODO
    }
}
```

Reply:
Makes sense, updated, let me know if that's what you were thinking